purchases/src/test/java/com/revenuecat/purchases/attributes/SubscriberAttributesDeviceCacheTests.kt

```diff
@@ -163,46 +180,249 @@ class SubscriberAttributesDeviceCacheTests {
 
         underTest.setAttributes(appUserID, mapOf(expectedAttributes.keys.toList()[1] to expectedAttributes.values.toList()[1]))
 
-        assertCapturedEqualsExpected(expectedAttributes)
+        assertCapturedEqualsExpected(mapOf(appUserID to expectedAttributes))
     }
 
     @Test
     fun `clearing caches also clears the subscriber attributes`() {
-        underTest.clearCachesForAppUserID()
-        verify { mockEditor.remove(underTest.subscriberAttributesCacheKey("appUserID")) }
+        val expectedAttributes = mapOf(
+            "tshirtsize" to SubscriberAttribute("tshirtsize", "L", isSynced = true)
+        )
+        mockNotEmptyCache(expectedAttributes)
+        underTest.clearCachesForAppUserID(appUserID)
+        verify {
+            mockEditor.putString(
+                "com.revenuecat.purchases.$apiKey.subscriberAttributes",
+                JSONObject().also {
+                    it.put("attributes", JSONObject())
+                }.toString()
+            )
+        }
+    }
+
+    @Test
+    fun `Given there are some unsynced attributes for multiple users`() {
```

Comment:
I don't quite understand the meaning of the phrase

Reply:
You're not alone 😆  It looks like I didn't finish writing it 🤦